src/Cms/Media.php

```diff
@@ -57,7 +57,17 @@ public static function link(
 		}
 
 		// try to generate a thumb for the file
-		return static::thumb($model, $hash, $filename);
+		try {
+			return static::thumb($model, $hash, $filename);
+		} catch (Throwable $e) {
```

Comment:
Not sure, but maybe we could restrict this to just the `NotFoundException`. Then all actual errors will still produce HTTP 500 and the fatal page, while the non-existing thumbs will return the error page with HTTP 404. With this change, we also wouldn't need to differentiate between debug and non-debug.

Maybe I'm missing something though, so feel free to correct me :)

Reply:
I like the idea to really look for the specific NotFoundException

Reply:
@lukasbestle I'm a bit confused. Actually `static::thumb` already throws `NotFoundException` but page status code is 500 🤷‍♂️ 
https://github.com/getkirby/kirby/blob/fix/6446-non-existing-file-links/src/Cms/Media.php#L122-L123

Can you explain what you have in mind, please?

Reply:
Nvm, I get it. I'll implement.